File: meme_inator_back/apps/profiles/application/usecases/replace_my_profile_usecase.py
from typing import Dict, Any
from uuid import UUID
from apps.profiles.domain.entities.profile_entity import ProfileEntity
from apps.profiles.domain.usecases.ireplace_my_profile_usecase import IReplaceMyProfileUsecase
from apps.profiles.domain.irepositories.i_profile_repository import IProfileRepository
from core.results import Result

class ReplaceMyProfileUsecase(IReplaceMyProfileUsecase):
    def __init__(self, profile_repo: IProfileRepository):
        self.profile_repo = profile_repo

    def execute(self, user_id: UUID, full_data: Dict[str, Any]) -> Result[ProfileEntity]:
        return self.profile_repo.replace_my_profile(user_id, full_data)

File: meme_inator_front/lib/features/profiles/domain/usecases/replace_my_profile_usecase.dart
import 'package:meme_inator_front/core/iusecase.dart';
import 'package:meme_inator_front/core/results.dart';
import 'package:meme_inator_front/features/profiles/domain/entities/profile_entity.dart';
import 'package:meme_inator_front/features/profiles/domain/interfaces/iprofile_repository.dart';
import 'package:uuid/uuid.dart';

class ReplaceMyProfileInput {
  final Uuid userId;
  final Map<String, dynamic> profileData;

  const ReplaceMyProfileInput({
    required this.userId,
    required this.profileData,
  });
}

class ReplaceMyProfileUsecase
    extends IUseCase<ReplaceMyProfileInput, ProfileEntity> {
  final IProfileRepository profileRepo;

  ReplaceMyProfileUsecase(this.profileRepo);

  @override
  Future<Result<ProfileEntity>> execute({
    required ReplaceMyProfileInput valObj,
  }) async {
    throw UnimplementedError('ReplaceMyProfileUsecase.execute not implemented');
  }
}
